File: backend/api-gateway/health-check.js
const http = require('http');

const options = {
  host: 'localhost',
  port: process.env.PORT || 3000,
  path: '/health',
  timeout: 2000,
  method: 'GET'
};

const healthCheck = http.request(options, (res) => {
  console.log(`HEALTHCHECK STATUS: ${res.statusCode}`);
  if (res.statusCode === 200) {
    process.exit(0);
  } else {
    process.exit(1);
  }
});

healthCheck.on('error', (err) => {
  console.error('HEALTHCHECK ERROR:', err.message);
  process.exit(1);
});

healthCheck.on('timeout', () => {
  console.error('HEALTHCHECK TIMEOUT');
  healthCheck.destroy();
  process.exit(1);
});

healthCheck.end();

File: backend/ai-service/health-check.py
import urllib.request
import urllib.error
import sys
import os

def health_check():
    port = os.environ.get('PORT', '5000')
    url = f'http://localhost:{port}/health'
    
    try:
        response = urllib.request.urlopen(url, timeout=2)
        status_code = response.getcode()
        print(f'HEALTHCHECK STATUS: {status_code}')
        
        if status_code == 200:
            sys.exit(0)
        else:
            sys.exit(1)
            
    except urllib.error.URLError as e:
        print(f'HEALTHCHECK ERROR: {e}')
        sys.exit(1)
    except Exception as e:
        print(f'HEALTHCHECK ERROR: {e}')
        sys.exit(1)

if __name__ == '__main__':
    health_check()
